docs/source/problems.rst

```diff
@@ -15,37 +15,43 @@ Problems
 Built-in problems
```

Comment:
Maybe we could nest `built-in problems` in the parent key, no? Either way, the current structure is a bit sloppy: let's improve on it! 

![image](https://user-images.githubusercontent.com/35239244/211955900-7c2ff4ce-53ba-4bfe-8a45-752c600d24f3.png)